dev/VSIX/Shared/WizardImplementation.cs

```diff
@@ -61,48 +62,58 @@ public void ProjectFinishedGenerating(Project project)
                 }
             }
         }
-        private async Task InstallNuGetPackageAsync(IVsPackageInstaller installer, string packageId)
+        private async Task InstallNuGetPackagesAsync()
         {
-            await Task.Run(() =>
+            await ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
-                try
-                {
-                    installer.InstallPackage(null, _project, packageId, "", false);
-                    // If there's any CPU-bound work, it will be done on the background thread.
-                }
-                catch (Exception ex)
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                int canceled; // This variable will store the status after the dialog is closed
+
+                // Start the package installation task but do not await it here
+                var installationTask = StartInstallationAsync();
+
+                // Start the threaded wait dialog
+                _waitDialog.StartWaitDialog("Installing NuGet packages", "Please wait while NuGet packages are being installed to your project...", null, null, "Operation in progress...", 0, false, true);
```

Comment:
We should follow the VS model and pass null for the caption.  It will default to Microsoft Visual Studio.  This will keep us in sync with the Creation dialog that precedes it.

Reply:
The screenshot I saw only had two texts, not "Operation in progress..." - where does that appear? do we need it? (simpler is best)

I would go with this for szWaitMessage:
"Installing NuGet packages into project..."

Reply:
thanks @ghost1372 - @ShashankNay I'm okay with retaining